### PR TITLE
Mark exercise answered based on user answers

### PR DESCRIPTION
**Pull request by bryanjenningz**
Changes:
* User's answers get saved on the frontend.
* Exercise preview get displayed differently based on whether the user answered that question.
* Fixed exercise list styling bug that occurred when there were only 1 or 2 exercises.

How to test:
* Create a DOJO module at /admin/lessons/js0/modules
* Create a DOJO exercise at /curriculum/js0/mentor
* Go to the DOJO exercises page at /exercises/js0
* Click on "SOLVE EXERCISES" button.
* Answer an exercise.
* Click back arrow and verify that the exercise is now marked as answered.

This pull request doesn't actually save the user's answers. If they refresh the page then their answers are cleared. We will store the user's answers on the backend in a subsequent pull request.

This pull request is a part of https://github.com/garageScript/c0d3-app/issues/2253 and https://github.com/garageScript/c0d3-app/issues/2251

### CODE REVIEW

File: pages/exercises/[lessonSlug].tsx [hunk 1]
```diff
@@ -9,36 +9,21 @@ import Error, { StatusCode } from '../../components/Error'
...
-  { moduleName: 'Variables', state: 'ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'NOT ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'NOT ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'NOT ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'NOT ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'NOT ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'NOT ANSWERED', problem: exampleProblem },
-  { moduleName: 'Variables', state: 'ANSWERED', problem: exampleProblem }
-]
-
```
Comment (bryanjenningz) [inline pages/exercises/[lessonSlug].tsx:35]: Removed the mock exercise previews, we're now using the backend exercises for the previews.

File: pages/exercises/[lessonSlug].tsx [hunk 2]
```diff
@@ -9,36 +9,21 @@ import Error, { StatusCode } from '../../components/Error'
...
 const Exercises: React.FC<QueryDataProps<GetExercisesQuery>> = ({
   queryData
 }) => {
   const { lessons, alerts, exercises } = queryData
   const router = useRouter()
   const [exerciseIndex, setExerciseIndex] = useState(-1)
+  const [userAnswers, setUserAnswers] = useState<Record<number, string>>({})
```
Comment (bryanjenningz) [inline pages/exercises/[lessonSlug].tsx:25]: We store the user's answers on the frontend for now. They go away when you refresh the page. This is good for testing. In the next pull request, we'll get the user's answers from the backend.

File: pages/exercises/[lessonSlug].tsx [hunk 3]
```diff
@@ -78,12 +64,21 @@ const Exercises: React.FC<QueryDataProps<GetExercisesQuery>> = ({
           lessonTitle={currentLesson.title}
           hasPrevious={exerciseIndex > 0}
           hasNext={exerciseIndex < currentExercises.length - 1}
+          submitUserAnswer={(userAnswer: string) =>
+            setUserAnswers({ ...userAnswers, [exercise.id]: userAnswer })
```
Comment (bryanjenningz) [inline pages/exercises/[lessonSlug].tsx:68]: Here we add the user's answer to our userAnswers object which stores all the user's answers by exercise ID.

File: pages/exercises/[lessonSlug].tsx [hunk 4]
```diff
@@ -199,14 +204,16 @@ const ExerciseList = ({
...
       <div className={styles.exerciseList__container}>
-        {mockExercisePreviews.map((exercisePreview, i) => (
+        {exercises.map((exercise, i) => (
           <ExercisePreviewCard
             key={i}
-            moduleName={exercisePreview.moduleName}
-            state={exercisePreview.state}
-            problem={exercisePreview.problem}
+            moduleName={exercise.moduleName}
+            state={exercise.userAnswer === null ? 'NOT ANSWERED' : 'ANSWERED'}
```
Comment (bryanjenningz) [inline pages/exercises/[lessonSlug].tsx:211]: Right now there are only 2 states, but I think there should be a third state in the future called "ANSWERED INCORRECTLY" or something similar. Maybe we can make "ANSWERED INCORRECTLY" red and then "NOT ANSWERED" can be gray or some neutral color.

**Review by flacial — APPROVED** [review]

**Pull request merged**